sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -1716,31 +1714,10 @@ static <T> PCollection<Iterable<T>> batchElements(
               .apply(Values.create());
     } else {
       iterables =
-          input.apply(
-              ParDo.of(
-                  new DoFn<T, Iterable<T>>() {
-                    @Nullable List<T> outputList;
-
-                    @ProcessElement
-                    public void process(ProcessContext c) {
-                      if (outputList == null) {
-                        outputList = new ArrayList<>();
-                      }
-                      outputList.add(c.element());
-                      if (outputList.size() > batchSize) {
-                        c.output(outputList);
-                        outputList = null;
-                      }
-                    }
-
-                    @FinishBundle
-                    public void finish(FinishBundleContext c) {
-                      if (outputList != null && outputList.size() > 0) {
-                        c.output(outputList, Instant.now(), GlobalWindow.INSTANCE);
-                      }
-                      outputList = null;
-                    }
-                  }));
+          input
+              .apply(WithKeys.<String, T>of(""))
+              .apply(GroupIntoBatches.<String, T>ofSize(batchSize).withShardedKey())
```

Comment:
```suggestion
              .apply(GroupIntoBatches.<String, T>ofSize(batchSize)..withMaxBufferingDuration(Duration.millis(200)))
```

Reply:
this creates the same kind of per-batch grouping but no autosharding

Reply:
thanks for the detailed explanation. I am wondering what is the performance impact of changing it to GroupIntoBatches. Should we preserve the current implementation (a DoFn) used for batch, and use GroupIntoBatches  in streaming with/without autosharding?